Packages/cdms2/Lib/axis.py

```diff
@@ -735,10 +735,22 @@ def designateTime(self, persistent=0, calendar=None):
             if calendar is not None:
                 self.setCalendar(calendar, persistent)
 
+    # For isTime(), keep track of whether each id is for a time axis or not, for better performance.
+    # This dictionary is a class variable (not a member of any particular instance).
+    idtaxis = {}  # id:type where type is 'T' for time, 'O' for other
+
```

Comment:
@painter1 what happens if you have mutliple files with same axis id, but only some are actually time? Wouldn't this break?